DRYD-1133: Chronology Authority

mikejritter opened the pull request:
**What does this do?**
Creates a new Chronology Authority

**Why are we doing this? (with JIRA link)**
https://collectionspace.atlassian.net/browse/DRYD-1133
_Note: Specs are in DRYD-1134_

This new authority allows records to reference eras (e.g. archaeological, geological, historical, etc) and events (e.g. presidential terms, wars, olympic games, etc).

**How should this be tested? Do these changes have associated tests?**

Check that all tests pass first and there are no linting issues
* `npm run test`
* `npm run lint`

Then using a local instance of CollectionSpace verify functionality of the new authority
* `npm run devserver`
* Check that the UI matches the wireframes
* Create a Chronology Authority
* Associate various authorities to the new Chronology
* Reload the Chronology to verify everything saved
* Check search + advanced search works for Chronology

**Dependencies for merging? Releasing to production?**
PRs for adding Chronology to the application and services layers

**Has the application documentation been updated for these changes?**
No

**Did someone actually run this code to verify it works?**
@mikejritter ran and tested against a local instance

kspurgin commented:
@mikejritter - Can the vocabularies.js get `disableAltTerms: true` added to each vocabulary as pending in: https://github.com/collectionspace/cspace-ui.js/pull/152

Not knowing the order of how things will get merged in, I'm not sure how to best make sure doing that doesn't fall through the cracks

ray-lee commented on src/plugins/recordTypes/chronology/fields.js:
If the Associated Authorities fields are going to be on every authority record, I'd recommend putting them into a UI extension, so you don't have to define them repeatedly in every authority. That way we also won't need to have a ton of messages that are all the same, and probably would be translated the same.

mikejritter replied:
Yep, I didn't realized this was the intention until last week, so I have some adjustments to make to this and the application layer.

mikejritter commented on src/plugins/recordTypes/chronology/fields.js:
```diff
@@ -0,0 +1,1397 @@
+                view: {
+                  type: AutocompleteInput,
+                  props: {
+                    source: 'chronology/era,chronology/event,chronology/fieldcollection',
```
I must have missed this when looking at other authorities and just assumed it would be self referential.

mikejritter commented:
@ray-lee I updated this to correct the `termSource` source and move the associated authorities block into an extension. I also rebased against master so it should be good to merge assuming everything is ok.

ray-lee commented:
@mikejritter Looks like there are some test failures.

mikejritter commented:
@ray-lee Took me a minute to see what I was missing - I added the new extension to the context config creation and tests are passing now

ray-lee commented on src/plugins/extensions/associatedAuthority/fields.js:
```diff
@@ -0,0 +1,776 @@
+            view: {
+              type: TermPickerInput,
+              props: {
+                source: 'chronologypersonrelations',
```
Hmm, are these relation terms really going to be specific to each record type? If so, we can't really reuse this config, and it shouldn't be an extension -- which would be bad. We should clarify this.

ray-lee reviewed (approved):
I think this is fine to merge, but I do have a concern about the term lists used in the associated authority fields that we should figure out.
